fmp_server.py

```diff
@@ -40,6 +40,237 @@ def recvAnsConnection(self, packet_id, data, seq):
+                    data += user.pack()
+                    self.server.layer_broadcast(self.session, PatID4.NtcLayerIn,
+                                                data, seq)
+            else:
+                # Same Server, same Gate
+                if city_id:
+                    city = self.session.find_city(server_id, gate_id, city_id)
+                    with city.lock():
+                        if city.is_empty():
+                            self.sendAnsAlert(PatID4.AnsLayerJump,
+                                    "<LF=8><BODY><CENTER>City no longer exists.<END>",
+                                    seq)
+                            return
+                        elif city.is_full():
+                            self.sendAnsAlert(PatID4.AnsLayerJump,
+                                    "<LF=8><BODY><CENTER>Destination is full.<END>",
+                                    seq)
+                            return
+
+                        # Step 1: Leave current location
+                        while self.session.layer > 1:
+                            self.notify_layer_departure()
+                            self.session.layer_up()
+                        # Step 2: Join City
+                        self.session.layer_down(city_id)
+                else:
+                    # Step 1: Leave current location
+                    while self.session.layer > 1:
+                        self.notify_layer_departure()
+                        self.session.layer_up()
+        self.sendAnsLayerJump(seq)
+
+    def sendAnsLayerJump(self, seq):
+        """NtcLayerJumpGo packet.
+
+        ID: 64170200
+        JP: レイヤ予約移動実行返答
+        TR: Layer reservation move execution reply
+        """
+        self.send_packet(PatID4.AnsLayerJump, b"", seq)
+
+    def recvReqLayerJumpReady(self, packet_id, data, seq):
+        """AnsLayerJump packet.
+
+        ID: 64100200
+        JP: レイヤ移動返答（位置バイナリ）
+        TR: Layer move reply (position binary)
+    
+        Client asking permission to switch Servers.
+        """
+        length, unk1, server_id, unk2, gate_id, city_id = struct.unpack_from(">HIIHHH", data)
+        if self.session.get_server_full(server_id) or\
+            self.session.get_gate_full(server_id, gate_id) or\
+            (city_id and self.session.get_city_full(server_id, gate_id, city_id)):
+            self.sendAnsAlert(PatID4.NtcLayerJumpReady,
+                        "<LF=8><BODY><CENTER>Destination is full.<END>",
+                        seq)
+            return
+        # Step 1: Leave current location
+        # ( continued in sendNtcLayerJumpReady() )
+        while self.session.layer > 0:
+            self.notify_layer_departure()
+            self.session.layer_up()
+        self.sendNtcLayerJumpReady(seq)
+
+    def sendNtcLayerJumpReady(self, seq):
+        """ReqLayerJumpReady packet.
+
+        ID: 64160200
+        JP: レイヤ予約移動確認返答
+        TR: Layer reservation move confirmation reply
+        """
```

Comment:
```suggestion
    def sendNtcLayerJumpReady(self, seq):
        """NtcLayerJumpReady packet.

        ID: 64160200
        JP: レイヤ予約移動確認返答
        TR: Layer reservation move confirmation reply
        """
```
sic: I think this function was wrongly identified when I generated the constants so we should rename it `AnsLayerJumpReady` later at some point or add a comment stating that this function was wrongly named in game if that was the case.